index.rst

```diff
@@ -1,61 +1,263 @@
+In terms of throughput, DDS is considerably better than Kafka for smaller messages, though we obtain similar values for larger messages.
+
+Overall, our detailed study shows that Kafka would be a viable option for replacing DDS as the middleware technology in our system.
+For the full technical report see :tstn:`033`.
 
-.. Please do not modify tocdepth; will be fixed when a new Sphinx theme is shipped.
+Summary
+=======
 
-.. sectnum::
+After considerable effort fine tunning the DDS middleware configuration, we were finally able to obtain a stable system, that is capable of operating at large scale with low middleware-related failure rate.
+At the current advanced state of the project, which is approaching its final construction stages, one might be tempted to accept this part of the project as concluded.
 
-.. TODO: Delete the note below before merging new content to the main branch.
+Nevertheless, as we demonstrated, there are a number of issues hiding underneath that may pose significant problems in the future, or even be seen as violating system requirements.
 
-.. note::
+Overall our experience with DDS has been, to say the least, underwhelming.
+Even thought the technology is capable of achieving impressive throughput and latency, in reality, it proved to be extremely cumbersome and hard to manage and debug on large scale systems.
+On top of if all we also face a potentially end-of-life cycle of the adopted library, which makes the problem considerably worse.
 
-   **This technote is not yet published.**
+After extensive exploring different alternatives to the problem we propose a potential long term solution to the problem, namely, replace DDS by the already in-use Kafka.
+Our benchmarks shows that Kafka is able to fulfill our system throughput and latency requirements.
+We also shown that transitioning to Kafka would require minimum effort and minimum code refactor.
 
-   This technote gathers information about the history, current state and contemplates future evolutions of the Vera Rubin Observatory Control System (Rubin-OCS) Middleware. The middleware is the backbone of the Rubin-OCS. The highly distributed nature of the Rubin-OCS places tight constraints in terms of latency, availability and reliability for the middleware. Here we gather information to answer some common questions regarding technology choices, describe some of the in-house work done to obtain a stable system and document some of the concerns with the current state of the system, potential impacts for near-future/commissioning and future/operations. We also cover some of the work we have been doing to investigate alternative technologies and suggest some potential roadmaps for the future of the system.
+We also note that there are major advantages of transitioning to Kafka before the end of construction.
+To begin with, we take advantage of a "standing army", as developers are actively engaged with the system and motivated.
```

Comment:
I am VERY wary of using the "standing army" at the moment, as we have stuff piling up on our doorstep and not getting completed and folks are starting to complain.  They may read this and wonder "why hasn't TSSW done X for me" which will leave me with a lot of explaining to do - as well as everyone actually is very busy in multiple components - so there really isn't any "standing army".   You may want to state that we have the expertise on-hand during the construction phase and we want to take advantage of that before it disappears - which is more correct.
And if anyone feels if they are standing around, please let me know.  ;-)